### ISSUE TITLE
ess-r-package-eval-linewise problem on remote R session

### ISSUE THREAD

**Issue by dpritchLibre** [emacs-ess/ESS#950]
The `ess-r-package-eval-linewise` function does not work properly for me when running an R session on a remote machine through Tramp.  The difficulty that I have is that the function uses the Tramp filename such as e.g. `"/ssh:remote.server.name:/home/username/packagename"` rather than `"/home/username/packagename"`, the former of which R obviously doesn't understand.

This issue affects the `ess-r-devtools` family of functions.

The `ess-r-package-eval-linewise` function gets the path string through a call to `ess-r-package-info` which in turn gets the value from a call to `ess-r-package--find-package-path`.  It appears to me that somewhere along the way it might be desirable to drop the Tramp portion of the path string to enable the correct functionality on remote machines.

Two functions that might be helpful for such a task are `tramp-tramp-file-p` and `tramp-dissect-file-name`.  So for example, for a path string `path`, one could do the following:
```
(if (tramp-tramp-file-p path)
    (nth 6 (tramp-dissect-file-name path))
  path)
```
I would be happy to submit a PR inserting such a section of code if it were decided where the best place for it.  One thing that I don't have any understanding of though is how package dependencies (i.e. the dependency on tramp) in Emacs are specified and whether any code would need to be added in that regard.

**Comment by jabranham** [comment]
Thanks for the detailed bug report. I just pushed a patch for this. Let  me know if it doesn't fix the issue for you!

**Comment by dpritchLibre** [comment]
@jabranham Thanks very much for the patch for this issue.  Unfortunately, although I think it solves the initial problem, I believe that the change has now caused a new difficulty.

The call to `ess-r-package-info` from `ess-r-package-eval-linewise` is returning `'(nil)`.  It turns out that this is due to a call to `(ess-r-package--find-package-name path)` from `ess-r-package-info`, where the argument provided for `path` is the pathname of the package after stripping out the Tramp portion of the path string.  However, the `ess-r-package--find-package-name` function actually needs the full Tramp path string as the argument since as it turns out it looks at the contents of the DESCRIPTION file in the package to determine the package name (so in other words it is looking for the file on the local machine as opposed to the remote machine where in fact the file exists).

**Comment by lionel-** [comment]
Closed by #1034.
